core/src/test/java/com/linecorp/armeria/client/retry/RetryingClientTest.java

```diff
@@ -509,6 +510,9 @@ void honorRetryMapping() {
         assertThat(client.get("/502-always").aggregate().join().status())
                 .isEqualTo(HttpStatus.valueOf(502));
         assertThat(stopwatch.elapsed()).isBetween(Duration.ofSeconds(0), Duration.ofSeconds(2));
+
+        // We sent 3 logical requests so the retry mapping should have been only evaluated 3 times.
+        assertThat(retryMappingEvaluations.get()).isEqualTo(3);
```

Comment:
This new assert fails on the "old" code with 14 != 3.

Reply:
Would you mind testing `RetryingRpcClient` with Thrift clients?

Reply:
Done, PTAL.